handler/src/main/java/io/netty/handler/traffic/AbstractTrafficShapingHandler.java

```diff
@@ -512,20 +512,22 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg) throw
 
     /**
      * Method overridden in GTSH to take into account specific timer for the channel.
+     * @param ctx
      * @param wait the wait delay computed in ms
      * @param now the relative now time in ms
      * @return the wait to use according to the context
      */
-    long checkWaitReadTime(final ChannelHandlerContext ctx, long wait, final long now) {
+    protected long checkWaitReadTime(final ChannelHandlerContext ctx, long wait, final long now) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "ctx". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1172)